### DISCUSSION TITLE
[Help]: Custom theming

### DISCUSSION THREAD

**Discussion by yashranjan1** [prasanthrangan/hyprdots#892] [General]
I tried to create a custom them but it doesnt seemt to be applying to some apps like waybar. I read the documentation, added the custom theme folders, files and also added an entry into the ``` theme.ctl ``` file but i still cant get the theme to apply on all apps. Also when I try to switch my theme, the thumbnail of my custom theme is missing even though I provided it a path to my wallpaper. I tried to look for a similar issue but I could not find one. However, if I missed something it would be great if you could guide me.

**Comment by kRHYME7** [comment]
Can you send some screenshots? And for wallpapers it should be in ``` ~/.config/swww/$Theme_Name 

Also you need to run ``` ./create_cache.sh ```or ``` Hyprdots cache reload ``` to generate all thumbnails

**Comment by yashranjan1** [comment]
Yes thats where i saved my wallpaper. My themes name is Rain-Dark. I ran that command and the thumbnail was generated, thanks for that. However the waybar issues still remain. Here is a screenshot of the ```theme.ctl``` file 

![240217_12h24m53s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/113207367/2e584dce-3fc4-4070-acbc-af7b69098932)


I rechecked my config tree against the one provided in the wiki, and everything seems to be in order, all the files are there, names are consistent.

Another thing I observed is that even though I switch to my Rain-Dark theme, when i try to change the theme, Toko-Night is selected. I think this has something to do with me using all of tokyo-night's theme files as a template. Here is a screenshot of that behaviour.

![image](https://github.com/prasanthrangan/hyprdots/assets/113207367/397e2851-4d44-43ef-8765-5f27a60522e4)

My guess is I havent changed an important part of the config that is still pointing to the Tokyo-Night theme due to that being the template I have chosen for my new theme.  My new theme seems to be applying the config to kitty tho.

**Comment by kRHYME7** [comment]
The best way to test this is upload it on github or compress it into something so We could patch it to our system for testing.  I'll try to see the Underlying script for this part.




You can also run this ``` Hyprdots theme set ``` 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/4fd8c9f1-10fd-4f7f-9fef-96acf3181196)
then try to select another theme just to toggle some configs then back to Rainy-Dark



I see that on the last screenshot the gtk theme is still pointing to Tokyo. 
see if my hunch is correct using this comman ``` gsettings get org.gnome.desktop.interface gtk-theme | sed "s/'//g" ``` 
with that it should provide Rainy-Dark not Tokyo

If it outputs Tokyo-Night you should modify your Gtk_Rainy-Dark file.

This is the https://github.com/prasanthrangan/hyprdots-mod sample of minimum files you needed so we could patch it on our systems.

**Comment by yashranjan1** [comment]
I tried the command out and your hunch was right it did output Tokyo Night. Where should the Gtk_Rain-Dark file exist? 

Here is the minimum config, i couldnt find the Source/arcs folder anywhere, so Im assuming that I dont need to add that? Let me know if i do tho: here is a [link](https://github.com/yashranjan1/hyprdots-config) to my repo.

**Comment by kRHYME7** [comment]
Uhmm that is really the core Package needed. 

Think about it as the GTK package as the Heart of your configs,  so maybe instead of using the Tokyo-Night find the GTK theme for "Rainy-Dark" if this is not case you should modify the "Tokyo-Night" names.

Review this part and you can modify the files inside the GTK theme and change it to your THEME name. 
``` 
GTK apps

Download GTK3/4 themes and extract it to ~/.themes/My-Fav-Theme.
Themes are available in https://www.gnome-look.org/browse?cat=135&ord=rating.
You can also make your own gtk theme if you have time!

```


BTW to make the whole theme modular you should follow this file tree

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/f41c01d9-a70a-4dac-b5bb-6d60688db5cd)

****

**Comment by yashranjan1** [comment]
Where can i find it?

**Comment by kRHYME7** [comment]
https://github.com/prasanthrangan/hyprdots/discussions/653

You can follow this discussion I think this is almost done

You can follow this file tree 
https://github.com/Gigas002/hyprdots-sweet

**Comment by yashranjan1** [comment]
Ahhhh I understand, So what I did was manually create the files on directories for the theme but youre supposed to be using the patcher. That makes way more sense now. I was wondering how i could make this system modular to add custom themes, thats brilliant! I'll quickly change and update the repository. Thanks!

**Comment by kRHYME7** [comment]
Have you check out how the GTK theme works? I think you should edit that manually (If you want to change the theme name ) and decompress that file to ~/.themes. (the patcher will do it for you as long as the name is the Theme_Name)

Example here are the themes I have
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/0296a69e-95ad-4a13-9736-dc820a70734f)

Try to duplicate Tokyo-Night then rename the copy into your Theme_name

Mandatory files
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/a7d1f30a-e12d-4d31-92ba-852d90f4b0f5)
This one are the target directory for the ``` Gtk_Theme_Name ``` and others. 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/8a284917-daa8-4a5b-b718-aa99d6cdec16)

**Comment by kRHYME7** [comment]
I'm not good at ricing but here's my discord maybe I can help you with how this works. ``` khing0v0 ```

**Comment by yashranjan1** [comment]
Thanks! I have added you.

You can check my [repo](https://github.com/yashranjan1/hyprdots-config) . I have updated the config according to the template. The Gtk theme I downloaded seems to have an icon pack, so would i still need a separate icon theme file?

Edit: Most files havent been changed yet, I was still working on waybar, they should all be identical to Tokyo-night, the only configs  that differ a littl would be waybar and kitty.

**Comment by kRHYME7** [comment]
Let's continue this discussion on your repo then. Also I will move this to the Discussion as I think this suits it there.
